pkg/unshare/unshare_linux.go

```diff
@@ -76,6 +78,23 @@ func getRootlessGID() int {
 	return os.Getegid()
 }
 
+func isSetID(path string, modeid fs.FileMode, capid capability.Cap) (bool, error) {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	mode := info.Mode()
+	if mode&modeid == modeid {
+		return true, nil
+	}
+	cap, err := capability.NewFile(path)
```

Comment:
The docs for `github.com/syndtr/gocapability/capability` say `NewFile()` is deprecated, and to use `NewFile2()` and then `cap.Load()` instead.

Reply:
I had that on a previous version.